tests/basic/bug80384.phpt

```diff
@@ -0,0 +1,26 @@
+--TEST--
+Bug #80384 large reads cause filters to internally buffer large amounts of memory
+
+--FILE--
+<?php
+/* First, create a file to read */
+$tmp_filename = __DIR__ . "/bug80384.tmp";
+$fp = fopen($tmp_filename, 'w');
+for ($i=0; $i<1024; $i++) {
+    fwrite($fp, str_repeat('ABCDEFGH', 1024));
+}
+fclose($fp);
+
+/* Stream the file through a filter */
+$fp = fopen($tmp_filename, 'r');
+$filter = stream_filter_append($fp, "string.rot13");
+fread($fp, 8 * 1024 * 1024);
+
+echo (memory_get_usage() < 1024 * 1024);
```

Comment:
The `echo` is okay, but a `var_dump` may be slightly better. Also, it may make sense not to assume a fixed limit, but rather to compare the `memory_get_usage()` before and after reading from the stream.